tools/environment/export.go

```diff
@@ -0,0 +1,450 @@
+	return true
+}
+
+func (ex *Exporter) writeObject(dir, filename string, object any, extract ExtractMask, strip StripMask) error {
+	m, ok := object.(map[string]any)
+	if !ok {
+		return nil
+	}
+
+	if strip&StripVersion != 0 {
+		delete(m, "version")
+	}
+	if strip&StripID != 0 {
+		delete(m, "id")
+	}
+	if strip&StripModifiedTime != 0 {
+		delete(m, "modifiedTime")
+	}
+	if strip&StripProjectName != 0 {
+		delete(m, "projectName")
+	}
+
+	path, err := ex.ensurePath(dir)
+	if err != nil {
+		return err
+	}
+
+	fullpath := filepath.Join(path, filename+".json")
+
+	if ex.Extract&extract != 0 {
+		if err := ex.extractAssets(fullpath, m); err != nil {
+			return err
+		}
+	}
+
+	bytes, err := json.MarshalIndent(m, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	if err := os.WriteFile(fullpath, bytes, 0644); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (ex *Exporter) writeFlows(arr any) {
+	flowIds := []string{}
+	if flows, ok := arr.([]any); ok {
+		for i := range flows {
+			if object, ok := flows[i].(map[string]any); ok {
+				if flow, ok := object["flow"].(map[string]any); ok {
+					if flowId, ok := flow["id"].(string); ok {
+						flowIds = append(flowIds, flowId)
+						ex.writeFlow(flowId, object)
+					}
+				}
+			}
+		}
+	}
+	if len(flowIds) > 0 {
+		sort.Strings(flowIds)
+		list := map[string]any{"flows": flowIds}
+		ex.writeObject("flows", "flows", list, ExtractNone, StripNone)
+	}
+}
+
+func (ex *Exporter) writeFlow(flowId string, object map[string]any) {
+	ex.ensurePath("flows")
+	if ex.Split&SplitFlows != 0 {
+		dir := filepath.Join("flows", flowId)
+		ex.writeObject(dir, "flow", object["flow"], ExtractFlows, StripVersion|StripModifiedTime)
+		if screens, ok := object["screens"].([]any); ok {
+			for j := range screens {
+				if screen, ok := screens[j].(map[string]any); ok {
+					ex.writeObject(dir, fmt.Sprintf("screen-%d", j+1), screen, ExtractFlows, StripVersion)
+				}
+			}
+		}
+	} else {
+		if flow, ok := object["flow"].(map[string]any); ok {
+			delete(flow, "version")
+			delete(flow, "modifiedTime")
+		}
+		if screens, ok := object["screens"].([]any); ok {
+			for j := range screens {
+				if screen, ok := screens[j].(map[string]any); ok {
+					delete(screen, "version")
+				}
+			}
+		}
+		ex.writeObject("flows", flowId, object, ExtractFlows, StripNone)
+	}
+}
+
+func (ex *Exporter) writeTheme(object any) {
+	if ex.Split&SplitStyles != 0 {
+		if theme, ok := object.(map[string]any); ok {
+			if template, ok := theme["cssTemplate"].(map[string]any); ok {
+				ex.writeObject("styles", "dark", template["dark"], ExtractStyles, StripNone)
+				ex.writeObject("styles", "light", template["light"], ExtractStyles, StripNone)
+			}
+		}
+	} else {
+		ex.writeObject("/", "theme", object, ExtractStyles, StripModel)
+	}
+}
+
+func (ex *Exporter) ensurePath(dir string) (string, error) {
+	path := filepath.Join(ex.Root, dir)
+	if _, err := os.Stat(path); os.IsNotExist(err) {
```

Comment:
not critical, if err is a different error then `IsNotExist` you may want to return it